core/src/main/java/org/elasticsearch/percolator/PercolateDocumentParser.java

```diff
@@ -78,8 +75,8 @@ public ParsedDocument parse(PercolateShardRequest request, PercolateContext cont
         // Some queries (function_score query when for decay functions) rely on a SearchContext being set:
         // We switch types because this context needs to be in the context of the percolate queries in the shard and
         // not the in memory percolate doc
-        String[] previousTypes = context.types();
-        context.types(new String[]{PercolatorService.TYPE_NAME});
+        String[] previousTypes = context.getQueryShardContext().getTypes();
```

Comment:
there is some trickyness to this percolator stuff.. there where some test failures when I didn't pass or maybe due to other things but I can doublecheck. I'd rather keep it since @martijnvg refactors it anyways.